src/http/http-server-base.ts

```diff
@@ -4,33 +4,36 @@ import net from 'net'
 
 import {
   HttpIncomingMessage,
-  HttpJSONRequest as HttpJsonRequest,
+  HttpJsonRequest,
   HttpRequest,
   HttpRequestListener,
   HttpServerError,
-  HttpStringRequest as HttpTextRequest,
+  HttpTextRequest,
   readBody
 } from './http-common'
 
 export class HttpServerBase<T extends http.Server | https.Server> {
-  public listenPort = 0
+  public listenPort: number
   public listenUrl = ''
   private baseUrl: string
   protected httpServer: T
   protected requests: HttpRequest[] = []
```

Comment:
We can add the `requests` array as an option so you can reference it back to a request array in the WebServer class.